x/concentrated-liquidity/grpc_query.go

```diff
@@ -133,6 +134,34 @@ func (q Querier) Params(goCtx context.Context, req *types.QueryParamsRequest) (*
 	return &types.QueryParamsResponse{Params: q.Keeper.GetParams(ctx)}, nil
 }
 
+// TickLiquidityInBatches returns array of liquidity depths in the given range of lower tick and upper tick.
+// Note that the space between the ticks in the returned array would always be guaranteed spacing greater than given batch unit.
+func (q Querier) TickLiquidityInBatches(goCtx context.Context, req *types.QueryTickLiquidityInBatchesRequest) (*types.QueryTickLiquidityInBatchesResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	queryLimit := sdk.NewInt(tickLiquidityInBatchQueryLimit)
+	requestedAmt := req.UpperTick.Sub(req.LowerTick).Quo(sdk.NewInt(int64(req.BatchUnit)))
+	if queryLimit.GT(requestedAmt) {
```

Comment:
i think this should be `queryLimit.LT` or `requestedAmt.GT(queryLimit)`

Reply:
Fixed in https://github.com/osmosis-labs/osmosis/pull/4209/commits/55d014e68f53d037ef1ebc08eb81bacdeb0bde26, planning to add more tests for these once ack from FE side